src/theme/components/index.ts

```diff
@@ -0,0 +1,15 @@
+import { Components, Theme } from "@mui/material"
+import AppBarTheme from "./AppbarTheme"
+import BackdropTheme from "./BackdropTheme"
+import DialogTheme from "./DialogTheme"
+import TooltipTheme from "./TooltipTheme"
+import { merge } from "lodash"
+
+export default function componentsOverrides(theme: Theme): Components {
+  return merge(
+    AppBarTheme(theme),
+    DialogTheme(theme),
+    TooltipTheme,
```

Comment:
Do these theme functions need to be called like the ones above them, or is it okay to merge in raw functions like this?

Reply:
These themes can merge in one function. Because there are much component in MUI so, I thought it would be more clean to split theme file for each components.

Reply:
Why does `TooltipTheme` not have a parameter for theme? Will tooltip be the same style regardless of mode?

Reply:
It could have `theme` parameter, but it is not using theme for now. Actually `theme` parameter is used for using color and shadow. so `theme` parameter is optional and because of eslint, so I didn't give `theme` parameter for `TooltipTheme` for now. But we can add when we need.

Reply:
My question was why are some component themes passed in as raw functions (`ToolipTheme`) rather than as executed functions (`TooltipTheme()`) to return the theme object?

Reply:
@hammeiam  You are correct.  component themes should be passed as executed functions.